internal/server/teams.go

```diff
@@ -11,18 +12,47 @@ import (
 	"github.com/pkg/errors"
 )
 
-type team struct {
-	Rank         *int     `json:"rank,omitempty"`
-	RankingScore *float64 `json:"rankingScore,omitempty"`
+// teamHandler returns a handler to get general info for a specific team
+func (s *Server) teamHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if err := s.updateTeams(r.Context()); err != nil {
+			ihttp.Error(w, http.StatusInternalServerError)
+			go s.Logger.WithError(err).Error("unable to update teams data")
+			return
+		}
+
+		vars := mux.Vars(r)
+		teamKey := vars["teamKey"]
+
+		team, err := s.Store.GetTeam(r.Context(), teamKey)
+		if err != nil {
+			// 404 if teamKey isn't a real team
+			if _, ok := errors.Cause(err).(store.ErrNoResults); ok {
```

Comment:
Can you refactor this so it's an if and else if? E.g. `if _, ok := .... else if err != nil`